ivy_tests/test_ivy/test_frontends/test_numpy/test_indexing_routines/test_indexing_like_operations.py

```diff
@@ -150,6 +150,49 @@ def test_numpy_put_along_axis(
     )
 
 
+@handle_frontend_test(
+    fn_tree="numpy.putmask",
+    dtypes_a=helpers.dtype_and_values(
+        available_dtypes=helpers.get_dtypes(kind="numeric", prune_function=True),
+        min_num_dims=1,
+        max_num_dims=5,
+        min_dim_size=10,
+        max_dim_size=100,
+    ),
+    test_with_out=st.just(False),
+    number_positional_args=st.just(2),
+)
+def test_numpy_putmask(
+    dtypes_a,
+    test_flags,
+    frontend,
+    backend_fw,
+    fn_tree,
+    on_device,
+):
+    input_dtypes, a = dtypes_a
+    a = a[0]
+    ip_dtype = input_dtypes[0]
+    mask = helpers.np.random.choice([True, False], size=a.shape)
+    mask = mask.astype(ip_dtype)
+    values_size = helpers.np.random.randint(1, a.size + 1)
+    values = helpers.np.random.randint(-100, 100, size=values_size)
+    print()
+    values = values.astype(ip_dtype)
```

Comment:
This is against convention, you might find another way to achieve the same behavior (maybe shared dtypes)